Assignment6.Rmd

```diff
@@ -70,14 +70,39 @@ table(M2$certified, M2$predict1)
 
 table(M2$certified, M2$predict2)
 
+#The pruned tree has a lower error rate.
+
 ```
 
 ##Part III
 
 Choose a data file from the (University of Michigan Open Data Set)[https://github.com/bkoester/PLA/tree/master/data]. Choose an outcome variable that you would like to predict. Build two models that predict that outcome from the other variables. The first model should use raw variables, the second should feature select or feature extract variables from the data. Which model is better according to the cross validation metrics?
 
 ```{r}
-
+library(dplyr)
+library(tidyverse)
+student.course<-read.csv("student.course.csv",header = TRUE)
+student.course<-filter(student.course, DIV == "P")
+student.course$Grade_Level=ifelse(student.course$GPAO>3.3,"A",ifelse(student.course$GPAO>2.3,"B",ifelse(student.course$GPAO>1,"C","F")))
+
+#split data into train set and test set
+set.seed(19)
+trainrow <- sample(1:nrow(student.course), size=0.8*nrow(student.course))
+train_set <- student.course[trainrow,]
+test_set <- student.course[-trainrow,]
+
+#classification tree
+c.tree3<-rpart(Grade_Level ~ .-GPAO,train_set)
+printcp(c.tree3)
+post(c.tree3,file = "tree3.ps", title = "course 1")
+prediction<-predict(c.tree3,test_set)
+
+#pruned classification tree
+c.tree4<-rpart(Grade_Level ~ GRD_PTS_PER_UNIT + SUBJECT,train_set)
+printcp(c.tree4)
+post(c.tree3,file = "tree4.ps", title = "course 1")
+
+#According to the cross-validation metrics, the first model is slightly better.Since the xerror of first model is 0.74527 and the xerror of the second model is 0.74615, the first model has a slight edge over the second model. They both have three splits and four nodes.
```

Comment:
Nice job! Clear model comparison. It would be great if you could explain and justify why you choose the two features in the second model.